ariadne/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -494,3 +495,46 @@ findFirstUnique lastIdx paths = fromMaybe (error "Can't find a unique path!")
     . head $ dropWhile (`S.member` pathsSet) [lastIdx..]
   where
     pathsSet = V.foldr S.insert S.empty paths
+
+-- ^ This function derives a 3-level address using account index, change index
+--   and address index. The input key is should be the master key (not the key
+--   that was derived from purpose and coin type)
+deriveLvl3KeyPair ::
```

Comment:
I'd rename it to something like `deriveBip44KeyPair`, which is more descriptive than `deriveLvl3KeyPair`, in my opinion.

Reply:
This is what the name was initially, but since this function can actually derive non BIP-44 keys (if we pass a non-hardened account index to it, I renamed it.